code/datums/components/infective.dm

```diff
@@ -1,12 +1,78 @@
 /datum/component/infective
+	dupe_mode = COMPONENT_DUPE_ALLOWED
 	var/list/datum/disease/diseases //make sure these are the static, non-processing versions!
+	var/expire_time
+	var/min_clean_strength = CLEAN_WEAK
 
-/datum/component/infective/Initialize(list/datum/disease/_diseases)
-	RegisterSignal(COMSIG_MOVABLE_CROSSED, .proc/Infect)
-	diseases = _diseases
+/datum/component/infective/Initialize(list/datum/disease/_diseases, expire_in)
+	if(islist(_diseases))
+		diseases = diseases
+	else
+		diseases = list(_diseases)
+	if(expire_in)
+		expire_time = world.time + expire_in
```

Comment:
Why store this if you don't use it and once set it's unchangeable.

Reply:
In case something ever needs to know how much longer the component will last, which nothing does at the moment.

Reply:
Seems it would be better to keep reference to the timer.